src/lib/FreeC/Pass/EtaConversionPass.hs

```diff
@@ -142,10 +272,24 @@ etaConvertSubExprs' expr = do
   let Just expr' = replaceChildTerms expr children'
   return expr'
 
+
 -------------------------------------------------------------------------------
 -- Arity                                                                     --
 -------------------------------------------------------------------------------
 
+-- | Find the minimum number of missing arguments (the arity of the expressions) among the alternatives
+--   for the right-hand side of a function declaration. 
+--   All non-zero arities on the right-hand side of a function declaration 
+--   have the same arity.
+findMinMissingArguments :: IR.Expr -> Converter Int
+findMinMissingArguments (IR.If _ _ e1 e2 _) =
+  minimumM (map findMinMissingArguments [e1, e2])
+findMinMissingArguments (IR.Case _ _ alts _) =
+  minimumM $ map (findMinMissingArguments . IR.altRhs) alts
```

Comment:
We can avoid to add `minimumM` and a monad utility module by writing the following.

```haskell
minimum <$> mapM (findMinMissingArguments . IR.altRhs) alts
```

Reply:
Good point!

Reply:
I had a version using `liftM` as well but refrained from this suggestion afterwards ; ) So, yes, please go ahead with this suggestion!